File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.java
class Solution {
    public boolean check(int[] nums) {
        int n=nums.length;
        int p=0;
        for(int i=0;i<n;i++){
            if(i!=(n-1)){
            if(nums[i+1]<nums[i]){
                p++;
            }}
        }
        if(nums[0]<nums[n-1]){
            p++;
        }

        if(p>1){
            return false;
        }
        else return true;
    }
}

File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.cpp
class Solution {
public:
    bool check(vector<int>& nums) {
        int n= nums.size();
        int mx=nums[0]; int mx2=-1;
         int r=0;
        for (int i=1;i<n;i++){
            if(mx<=nums[i] && r==0) mx=nums[i];
            else{
                r=1;
                if(mx<nums[i] || mx2>nums[i] || nums[i]>nums[0]) return false;
                mx2=nums[i];
            }
        }
        return true;
    }
};